### ISSUE TITLE
[BUG] Series.where incorrect result

### ISSUE THREAD

**Issue by ayushdg** [rapidsai/cudf#3304]
**Describe the bug**
`Series.where(condition, replacement)` fails if there are duplicates in the series and the boolean condition for any one of those duplicates evaluates to `False`

**Steps/Code to reproduce bug**
```
>>> a= cudf.Series([1,1,1,2,3])
>>> a.where(cudf.Series([False,True, True, True, True]), 200)
0    200
1    200
2    200
3    2
4    3
dtype: int64
```
**Expected behavior**
```
0    200
1    1
2    1
3    2
4    3
dtype: int64
```

**Environment overview (please complete the following information)**
 - Environment location: Nightly
 - Method of cuDF install: Docker

**Additional context**
Under the hood `where` uses the `numerical_column.find_and_replace` method which is the same method invoked by `Series.replace`. This method takes values to replace and the replacement to fill it with. `Series.where` just applies the boolmask to get values in the original series that fail the condition, and then calls the `find_and_replace` with these values. This logic fails if there are duplicate values in the series, some passing the condition and some failing it.

**Comment by galipremsagar** [comment]
```python
>>> a= cudf.Series([1,1,1,2,3])
>>> a.where(cudf.Series([False,True, True, True, True]), 200)
0    200
1      1
2      1
3      2
4      3
dtype: int64
```
@ayushdg This seems to be fixed, maybe we can close the issue if you can confirm the same?

**Comment by ayushdg** [comment]
Yes. Resolved by #4367 closing!
